### ISSUE TITLE
UnifiedMap: Illegal item: MARKER (when live mode is disabled)

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#14852]
### how to reproduce:
- open non-UnifiedMap
- disable live mode
- go back to home screen and open UnifiedMap

### result
map crashes

### stacktrace

```
java.lang.IllegalArgumentException: Illegal item: MARKER!:,icon[null]:bm:99x99px, hash:78743199, angle:0.0, x/yAnchor:0.5/1.0,zLevel:9
    	at cgeo.geocaching.unifiedmap.geoitemlayer.GeoItemLayer.put(GeoItemLayer.java:256)
    	at cgeo.geocaching.unifiedmap.geoitemlayer.GeoItemLayer.put(GeoItemLayer.java:246)
    	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer.lambda$new$1$cgeo-geocaching-unifiedmap-layers-GeoItemsLayer(GeoItemsLayer.java:80)
    	at cgeo.geocaching.unifiedmap.layers.GeoItemsLayer$$ExternalSyntheticLambda1.onChanged(Unknown Source:8)
    	at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
    	at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
    	at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
    	at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
    	at android.os.Handler.handleCallback(Handler.java:938)
    	at android.os.Handler.dispatchMessage(Handler.java:99)
    	at android.os.Looper.loop(Looper.java:223)
    	at android.app.ActivityThread.main(ActivityThread.java:7656)
    	at java.lang.reflect.Method.invoke(Native Method)
    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
```

**Comment by eddiemuc** [comment]
This happens because a MARKER was passed with null coordinates. Due to zLevel=9 and a look into class LayerHelper I assume that a waypoint w/o coordinates was passed to the GeoItemLayer.

However, GeoItemLayer should not crash when illegal Geo-Items are passed to it. The underlying error is the same as for https://github.com/cgeo/cgeo/issues/14811. I introduced a fix for this with PR #14812. This fix was not applied on the code version where above stack trace was created (I can see that from the line numbers of GeoItemLayer class). Do you need to rebase your PR?

**Comment by moving-bits** [comment]
Very likely, thanks for cross-checking. I was co-working on #14797 when it happened. I updated my local branches before yesterday's meeting, but checked out the PR again after that... Let's close this issue.
